substrate/runtime/staking/src/lib.rs

```diff
@@ -86,18 +102,23 @@ impl<Hashing, AccountId> ContractAddressFor<AccountId> for Hashing where
 	}
 }
 
+// MaybeSerializeDebug is workaround for https://github.com/rust-lang/rust/issues/26925 . Remove when sorted.
 pub trait Trait: system::Trait + session::Trait {
 	/// The balance of an account.
-	type Balance: Parameter + SimpleArithmetic + Slicable + Default + Copy;
+	type Balance: Parameter + SimpleArithmetic + Slicable + Default + Copy + As<Self::AccountIndex> + As<usize>;
+	/// Function type to get the contract address given the creator.
 	type DetermineContractAddress: ContractAddressFor<Self::AccountId>;
+	/// Type used for storing an account's index; implies the maximum number of accounts the system
+	/// can hold.
+	type AccountIndex: Parameter + Member + Slicable + SimpleArithmetic + As<u8> + As<u16> + As<u32> + As<usize> + Copy;
```

Comment:
this is fairly dangerous because it can truncate

Reply:
?

Reply:
the `As` trait is being used without accounting for overflow

Reply:
where?